### ISSUE TITLE
Multiple trigger during startup

### ISSUE THREAD

**Issue by putian9935** [qnimble/quarto-lib#1]
Hi there, 

When I try to use the trigger feature you supplied, it is found that one trigger happens instantly after the trigger is setup. Consider this MWE: 

```{cpp}
int count = 0;
bool volatile trigged;

void setup() {
    while (!Serial);  // block serial so we can capture all
    Serial.begin(115200);
    // setup trigger
    triggerMode(1, INPUT);
    enableInterruptTrigger(1, RISING_EDGE, [] { trigged = true; });
}

void loop() {
    Serial.printf("a%d\n",count);
    delay(200);
    if (trigged) {
        trigged = false;
        Serial.println(count);
        ++count;
    }
}
```

The trigger port 1 is connected to a pulse generator that is idle (no pulse) at this moment. Since there is no trigger, I would expect that the ``if(trigged)`` clause is never entered, thus every line this device outputs should be "a0". 

However, the actual output looks like 
```
a0
0
a1
a1
...
```
You can see that there is in fact a trigger as the program starts.

**Comment by putian9935** [comment]
To solve it, I notice the ``enableInterruptTrigger`` is different from Teensy's ``attachInterrupt`` in that the trigger pin(``TRIGGERx_IMR``) is enabled all the way during the trigger setup. After Teensy's implementation idea is adopted, the problem is gone.

**Comment by ben-qnimble** [comment]
Thanks for reporting and finding the fix. I've merged your fix and it is in the just-released 1.8.2. You may need to restart the Arduino-IDE for it to see the that 1.8.2 is available.
